pyomo/contrib/doe/fim_doe.py

```diff
@@ -268,9 +283,9 @@ def SP_measure_name(self, j, t,scenario_all=None, p=None, mode=None, legal_t=Tru
                 if type(self.name_and_index[measure_name][0]) is str:
```

Comment:
This function needs comments before each if and else line. I can probably figure it out with enough time. But a few good comments prevent me from needing to guess.